src/Parser/Core/Modules/AlwaysBeCasting.js

```diff
@@ -135,16 +153,57 @@ class AlwaysBeCasting extends Module {
   }
   applyActiveBuff(event) {
     const spellId = event.ability.guid;
-    const hasteGain = this.constructor.HASTE_BUFFS[spellId];
+    let hasteGain = undefined;
+    
+    //Added check so it gets the correct one, and adds stacks if necessary
+    if (this.constructor.HASTE_BUFFS[spellId]){
+      hasteGain = this.constructor.HASTE_BUFFS[spellId];
+    }
+    else if (this.constructor.STACKABLE_HASTE_BUFFS[spellId]){
+      hasteGain = this.constructor.STACKABLE_HASTE_BUFFS[spellId].Haste(this.combatants.selected);
+      this.constructor.STACKABLE_HASTE_BUFFS[spellId].CurrentStacks ++;
+    }
+
     if (hasteGain) {
       this.applyHasteGain(hasteGain);
 
       debug && console.log(`ABC: Current haste: ${this.currentHaste} (gained ${hasteGain} from ${spellId})`);
     }
   }
+  //Added apply buff stack event for stackable haste buffs
+  applyBuffStack(event) {
+    const spellId = event.ability.guid;
+    const stackInfo = this.constructor.STACKABLE_HASTE_BUFFS[spellId];
+    let hasteGain = undefined;
+
+    if (stackInfo){
+      hasteGain = stackInfo.Haste(this.combatants.selected);
+    }
+
+    if (hasteGain) {
+      //Only add haste stack if max stacks not already reached
+      if (stackInfo.MaxStacks === 0 || stackInfo.CurrentStacks < stackInfo.MaxStacks){
+        this.applyHasteGain(hasteGain);
+        stackInfo.CurrentStacks ++;
+      }
+
+      debug && console.log(`ABC: Current haste: ${this.currentHaste} (gained ${hasteGain} from ${spellId})`);
+    }
+  }
   removeActiveBuff(event) {
     const spellId = event.ability.guid;
-    const hasteGain = this.constructor.HASTE_BUFFS[spellId];
+    let hasteGain = undefined;
+    
+    //Added check so it gets the correct one
+    if (this.constructor.HASTE_BUFFS[spellId]){
+      hasteGain = this.constructor.HASTE_BUFFS[spellId];
```

Comment:
`let hasteGain = this.constructor.HASTE_BUFFS[spellId] || undefined;`

This will let you get rid of the if check, the variable will fall back to undefined if the key doesn't exist on the HASTE_BUFFS object.